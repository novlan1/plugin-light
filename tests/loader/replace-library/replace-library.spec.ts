import replaceLibrary from '../../../loader/replace-library';

jest.mock('loader-utils', () => ({
  getOptions: jest.fn().mockReturnValue({
    replaceLibraryList: [
      {
        from: 'vant',
        to: 'src/common/tools/fake-library/vant/index.js',
        exact: false,
      },
      {
        from: 'tim-js-sdk',
        to: 'src/common/tools/fake-library/tim-js-sdk/index.js',
        exact: false,
      },
      {
        from: '@tencent/nes-tim',
        to: 'src/common/tools/fake-library/nes-tim/index.js',
        exact: false,
      },
      {
        from: 'swiper',
        to: 'src/common/tools/fake-library/vant/index.js',
        exact: false,
      },
      {
        from: 'lodash-es',
        to: 'src/common/tools/fake-library/vant/index.js',
        exact: false,
      },
    ],
  }),
}));

describe('replaceLibrary', () => {
  it('replaceLibrary', () => {
    const source = `
import nesTim from '@tencent/nes-tim';
import { Swipe, SwipeItem, Tab, Tabs } from 'vant';
import TIM from 'tim-js-sdk';
import '@tencent/nes-tim/lib/nes-tim.css';
import Swiper2, { Navigation, Pagination } from 'swiper';
import get from 'lodash-es/get';
import { post } from 'lodash-es';
    `;
    process.env.VUE_APP_PLATFORM = 'mp-weixin';
    expect(replaceLibrary(source)).toMatchSnapshot();
  });
});
