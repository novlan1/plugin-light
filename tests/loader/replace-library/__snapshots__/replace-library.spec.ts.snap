// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaceLibrary replaceLibrary 1`] = `
"
import nesTim from 'src/common/tools/fake-library/nes-tim/index.js';
import { Swipe, SwipeItem, Tab, Tabs } from 'src/common/tools/fake-library/vant/index.js';
import TIM from 'src/common/tools/fake-library/tim-js-sdk/index.js';
import 'src/common/tools/fake-library/nes-tim/index.js';
import Swiper2, { Navigation, Pagination } from 'src/common/tools/fake-library/vant/index.js';
import get from 'src/common/tools/fake-library/vant/index.js';
import { post } from 'src/common/tools/fake-library/vant/index.js';
    "
`;
